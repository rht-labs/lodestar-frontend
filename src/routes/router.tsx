import React from 'react';
import {
  Switch,
  Route,
  Redirect,
} from 'react-router-dom';

import { FeatureRequest } from '../components/feature_request';
import { PrivateRoute } from '../components/authentication/private_route';
import { CallbackHandler } from '../components/authentication/callback_handler';
import { EngagementPane } from './engagement_pane';
import { Admin } from './admin';
import { Dashboard } from './dashboard';
import { EngagementFormProvider } from '../context/engagement_form_context';
import { UnauthorizedPage } from './unauthorized';
import LogoutPage from './logout'

function _OMPRouter() {

  return (
    <Switch>
      <PrivateRoute
        exact
        path="/"
        component={() => {
          return (
            <Dashboard />
          );
        }}
      />
      <PrivateRoute
        exact
        path="/engagements"
        component={() => {
          return (
            <EngagementFormProvider>
              <EngagementPane />
            </EngagementFormProvider>
          );
        }}
      />
      <PrivateRoute
        exact
        path="/admin"
        component={() => {
          return (
            <Admin />
          );
        }}
      />
      <PrivateRoute path="/private" component={() => <Redirect to="/" />} />
      <Route path="/logout" component={() => <LogoutPage />} />
      <Route path="/feature-request" component={FeatureRequest} />
      <Route path="/auth_callback" component={CallbackHandler} />
      <Route path="/unauthorized" component={UnauthorizedPage} />
    </Switch>
  );
}

export const OMPRouter = React.memo(_OMPRouter);