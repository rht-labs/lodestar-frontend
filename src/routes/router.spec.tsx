import React from 'react';
import { render } from '@testing-library/react';
import { LodestarRouter } from './router';
import { MemoryRouter } from 'react-router';

describe('Router component', () => {
  test('renders correctly', () => {
    expect(
      render(
        <MemoryRouter>
          <LodestarRouter />
        </MemoryRouter>
      )
    ).toMatchSnapshot();
  });
});
