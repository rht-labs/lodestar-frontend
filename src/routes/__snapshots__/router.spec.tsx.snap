// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Router component renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="landing-page"
      >
        <div
          class="pf-c-page"
        >
          <header
            class="pf-c-page__header"
          >
            <div
              class="pf-c-page__header-brand"
            >
              <a
                class="pf-c-page__header-brand-link"
              >
                <span
                  class="logo-text"
                >
                  LodeStar
                </span>
              </a>
            </div>
            <div
              class="pf-c-toolbar"
              id="pf-random-id-0"
            >
              <a
                href="/app"
              >
                <button
                  aria-disabled="false"
                  class="pf-c-button pf-m-primary"
                  data-cy="get_started_button"
                  data-ouia-component-id="OUIA-Generated-Button-primary-1"
                  data-ouia-component-type="PF4/Button"
                  data-ouia-safe="true"
                  type="button"
                >
                  Get Started
                </button>
              </a>
              <div
                class="pf-c-toolbar__content pf-m-hidden"
                hidden=""
              >
                <div
                  class="pf-c-toolbar__group"
                />
              </div>
            </div>
          </header>
          <main
            class="pf-c-page__main"
            tabindex="-1"
          >
            <div
              class="jumbotron-container"
              style="height: 100vh;"
            >
              <div
                class="info-video"
              >
                <div
                  class="aspect-ratio--16x9"
                >
                  <iframe
                    allow="autoplay *; fullscreen *; encrypted-media *"
                    class="aspect-ratio-object"
                    frameborder="0"
                    id="kaltura_player"
                    sandbox="allow-forms allow-same-origin allow-scripts allow-top-navigation allow-pointer-lock allow-popups allow-modals allow-orientation-lock allow-popups-to-escape-sandbox allow-presentation allow-top-navigation-by-user-activation"
                    src="https://cdnapisec.kaltura.com/p/2300461/sp/230046100/embedIframeJs/uiconf_id/42569541/partner_id/2300461?iframeembed=true&playerId=kaltura_player&entry_id=1_2h6lcr8p&flashvars[streamerType]=auto&flashvars[localizationCode]=en&flashvars[leadWithHTML5]=true&flashvars[sideBarContainer.plugin]=true&flashvars[sideBarContainer.position]=left&flashvars[sideBarContainer.clickToClose]=true&flashvars[chapters.plugin]=true&flashvars[chapters.layout]=vertical&flashvars[chapters.thumbnailRotator]=false&flashvars[streamSelector.plugin]=true&flashvars[EmbedPlayer.SpinnerTarget]=videoHolder&flashvars[dualScreen.plugin]=true&flashvars[hotspots.plugin]=1&flashvars[Kaltura.addCrossoriginToIframe]=true&&wid=1_un5h0nqz"
                    title="Kaltura Player"
                  />
                </div>
              </div>
              <div
                class="background-image"
              />
              <div
                class="background-image-overlay"
              />
            </div>
          </main>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="landing-page"
    >
      <div
        class="pf-c-page"
      >
        <header
          class="pf-c-page__header"
        >
          <div
            class="pf-c-page__header-brand"
          >
            <a
              class="pf-c-page__header-brand-link"
            >
              <span
                class="logo-text"
              >
                LodeStar
              </span>
            </a>
          </div>
          <div
            class="pf-c-toolbar"
            id="pf-random-id-0"
          >
            <a
              href="/app"
            >
              <button
                aria-disabled="false"
                class="pf-c-button pf-m-primary"
                data-cy="get_started_button"
                data-ouia-component-id="OUIA-Generated-Button-primary-1"
                data-ouia-component-type="PF4/Button"
                data-ouia-safe="true"
                type="button"
              >
                Get Started
              </button>
            </a>
            <div
              class="pf-c-toolbar__content pf-m-hidden"
              hidden=""
            >
              <div
                class="pf-c-toolbar__group"
              />
            </div>
          </div>
        </header>
        <main
          class="pf-c-page__main"
          tabindex="-1"
        >
          <div
            class="jumbotron-container"
            style="height: 100vh;"
          >
            <div
              class="info-video"
            >
              <div
                class="aspect-ratio--16x9"
              >
                <iframe
                  allow="autoplay *; fullscreen *; encrypted-media *"
                  class="aspect-ratio-object"
                  frameborder="0"
                  id="kaltura_player"
                  sandbox="allow-forms allow-same-origin allow-scripts allow-top-navigation allow-pointer-lock allow-popups allow-modals allow-orientation-lock allow-popups-to-escape-sandbox allow-presentation allow-top-navigation-by-user-activation"
                  src="https://cdnapisec.kaltura.com/p/2300461/sp/230046100/embedIframeJs/uiconf_id/42569541/partner_id/2300461?iframeembed=true&playerId=kaltura_player&entry_id=1_2h6lcr8p&flashvars[streamerType]=auto&flashvars[localizationCode]=en&flashvars[leadWithHTML5]=true&flashvars[sideBarContainer.plugin]=true&flashvars[sideBarContainer.position]=left&flashvars[sideBarContainer.clickToClose]=true&flashvars[chapters.plugin]=true&flashvars[chapters.layout]=vertical&flashvars[chapters.thumbnailRotator]=false&flashvars[streamSelector.plugin]=true&flashvars[EmbedPlayer.SpinnerTarget]=videoHolder&flashvars[dualScreen.plugin]=true&flashvars[hotspots.plugin]=1&flashvars[Kaltura.addCrossoriginToIframe]=true&&wid=1_un5h0nqz"
                  title="Kaltura Player"
                />
              </div>
            </div>
            <div
              class="background-image"
            />
            <div
              class="background-image-overlay"
            />
          </div>
        </main>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
